### PR TITLE
Move journal.js into lib directory

### PR DESCRIPTION
**Pull request by stefwalter**
Combine server.js, journalctl.js and renderer.js appropriately, factor out code that's not reused. And then move the code and bundle it the way that it will be with webpack.

### CODE REVIEW

**Comment by dperpeet** [comment]
tests failing

**Comment by stefwalter** [comment]
Rebased. Fixed mustache file inclusion.

File: pkg/systemd/journal.js [hunk 1]
```diff
@@ -17,152 +17,544 @@
...
-            filler.stop();
-    }
-
-    function update_query() {
-        stop_query();
+    var module = { };
 
-        var match = [ ];
+    /**
+     * journal([match, ...], [options])
```
Comment (larsu) [inline pkg/systemd/journal.js]: Mismatches the actual function name `journalctl`.

File: pkg/systemd/journal.js [hunk 2]
```diff
@@ -17,152 +17,544 @@
...
+        var options = { follow: true };
+        for (var i = 0; i < arguments.length; i++) {
+            var arg = arguments[i];
+            if (typeof arg == "string") {
+                matches.push(arg);
+            } else if (typeof arg == "object") {
+                if (arg instanceof Array)
+                    matches.push.apply(matches, arg);
+                else
+                    $.extend(options, arg);
```
Comment (larsu) [inline pkg/systemd/journal.js]: Is it intentional that `journalctl` can be called with options before or between matches? Maybe add a `break` and ignore anything after options?

**Comment by larsu** [comment]
Oh apparently I looked at an old diff. What was your last change about?

**Comment by stefwalter** [comment]
> Oh apparently I looked at an old diff. What was your last change about?

A rebase. And another is needed now.

File: pkg/systemd/journal.js (hunk 2 above)
Comment (stefwalter) [inline pkg/systemd/journal.js]: Done, although none of this code was rewritten in this pull request. And checked all the callers.

File: pkg/systemd/journal.js (hunk 1 above)
Comment (stefwalter) [inline pkg/systemd/journal.js]: Done.

**Comment by larsu** [comment]
This unfortunately breaks the "Load more" button below logs. Its stuck at "Loading...".

Also, I get

    Error: Mismatched anonymous define() module

when going to two pages that use journal.js consecutively.

**Comment by stefwalter** [comment]
This should be fixed now.

**Pull request closed without merging**